Okta.Core/OktaHttpClient.cs

```diff
@@ -175,6 +176,24 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpRequestType requestTy
                     {
                         httpClient.DefaultRequestHeaders.Remove("Authorization");
                     }
+                    if (bAddXForwardedHeader)
+                    {
+                        string ipAddress = null;
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            content = content.Trim();
+                            if (content.StartsWith("{") && content.EndsWith("}"))
+                            {
+                                var obj = JObject.Parse(content);
+                                ipAddress = (string)obj["context"]["ipAddress"];
+                            }
+                        }
+
+                        if (ipAddress != null)
+                        {
+                            httpClient.DefaultRequestHeaders.Add("X-Forwarded-For", ipAddress);

```

Comment:
Assuming the developer reuses the `IOktaHttpClient` object, this will add the header value to **all** subsequent requests. I think it should only be added to the current request. To do that, we'd have to construct an `HttpRequestMessage`.